aptos-move/aptos-vm/src/move_vm_ext/session/user_transaction_sessions/abort_hook.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    move_vm_ext::{session::respawned_session::RespawnedSession, AptosMoveResolver, SessionId},
+    transaction_metadata::TransactionMetadata,
+    AptosVM,
+};
+use aptos_vm_types::{change_set::VMChangeSet, storage::change_set_configs::ChangeSetConfigs};
+use derive_more::{Deref, DerefMut};
+use move_core_types::vm_status::VMStatus;
+
+#[derive(Deref, DerefMut)]
+pub struct AbortHookSession<'r, 'l> {
+    #[deref]
+    #[deref_mut]
+    session: RespawnedSession<'r, 'l>,
+}
+
+impl<'r, 'l> AbortHookSession<'r, 'l> {
+    pub fn new(
+        vm: &'l AptosVM,
+        txn_meta: &'l TransactionMetadata,
+        resolver: &'r impl AptosMoveResolver,
+        prologue_change_set: VMChangeSet,
```

Comment:
probably should be called `previous_session_change_set` (also in other session files)

Reply:
Well, it's a choice between generalization and encoding the exact business logic via the naming -- for example, I do want to emphasize that AbortHookSession takes the prologue_change_set, instead of the result from a failed user session. If I'm more paranoid, I'd add `struct PrologueChangeSet(VMChangeSet)` but it seems too verbose and unnecessary.